File: java/DataFrame/src/main/java/coyote/dataframe/BooleanType.java
package coyote.dataframe;

import coyote.util.ByteUtil;

/** Type representing a boolean value */
public class BooleanType implements FieldType
{
  private static final int _size = 8;

  private final static String _name = "BOL";




  public boolean checkType( Object obj )
  {
    return obj instanceof Boolean;
  }




  public Object decode( byte[] value )
  {
    return new Boolean( ByteUtil.retrieveBoolean( value, 0 ) );
  }




  public byte[] encode( Object obj )
  {
    return ByteUtil.renderBoolean( ( (Boolean)obj ).booleanValue() );
  }




  public String getTypeName()
  {
    return _name;
  }




  public boolean isNumeric()
  {
    return false;
  }




  public int getSize()
  {
    return _size;
  }

}

File: csharp/DataFrame/Coyote/DataFrame/BooleanType.cs
#region license
// Copyright (c) 2006 Stephan D. Cote' - All rights reserved.
// 
// This program and the accompanying materials are made available under the 
// terms of the MIT License which accompanies this distribution, and is 
// available at http://creativecommons.org/licenses/MIT/
// *
// Contributors:
//   Stephan D. Cote 
//      - Initial API and implementation
#endregion
using System;
using System.Collections.Generic;

namespace Coyote.DataFrame {



    /// <summary>
    /// Type representing a boolean value 
    /// </summary>
    public class BooleanType : FieldType {
        private const int _size = 1;

        private const string _name = "BOL";


        public virtual string TypeName {
            get {
                return _name;
            }
        }




        public virtual bool IsNumeric {
            get {
                return false;
            }
        }




        public virtual int Size {
            get {
                return _size;
            }
        }




        public virtual bool CheckType( object obj ) {
            return obj is bool;
        }




        public virtual object Decode( byte[] value ) {
            return ByteUtil.RetrieveBoolean( value, 0 );
        }




        public virtual byte[] Encode( object obj ) {
            return ByteUtil.RenderBoolean( (bool)obj );
        }




        public virtual string StringValue( byte[] val ) {
            if ( val == null ) {
                return "";
            } else {
                object obj = Decode( val );
                if ( obj != null )
                    return obj.ToString();
                else
                    return "";
            }
        }

    }

}